server/src/server.ts

```diff
@@ -255,6 +259,14 @@ async function validatePerlDocument(textDocument: TextDocument): Promise<void> {
         nLog("Perl Critic Time: " + (Date.now() - start)/1000 + " seconds", settings);
         sendDiags({ uri: textDocument.uri, diagnostics: allNewDiags });
     }
+
+    const diagImports = await pImports;
+    documentDiags.set(textDocument.uri, diagImports); // May need to clear out old ones if a user changed their settings.
```

Comment:
This code is tricky. The navigator sends the compilation results first without waiting for the critic since the critic results can be slow (especially with custom non-optimized policies). I changed this around to group perlimports and critic together.